core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -657,17 +668,19 @@ class LogSegment private[log] (val log: FileRecords,
 object LogSegment {
 
   def open(dir: File, baseOffset: Long, config: LogConfig, time: Time, fileAlreadyExists: Boolean = false,
-           initFileSize: Int = 0, preallocate: Boolean = false, fileSuffix: String = ""): LogSegment = {
+           initFileSize: Int = 0, preallocate: Boolean = false, fileSuffix: String = "",
+           needsRecovery: Boolean = true): LogSegment = {
```

Comment:
It seems that needsRecovery should default to false?

Reply:
Yeah, I changed this but didn't push. Thanks